lib/cusparse/interfaces.jl

```diff
@@ -75,73 +76,171 @@ for (taga, untaga) in tag_wrappers, (wrapa, transa, unwrapa) in op_wrappers
+    gemm('N', 'N', one(T), A, B, 'O')
 end
-
-function Base.:(+)(A::CuSparseMatrixCSR, B::CuSparseMatrix)
-    csrB = CuSparseMatrixCSR(B)
-    return geam(one(eltype(A)), A, one(eltype(A)), csrB, 'O')
+function LinearAlgebra.:(*)(A::CuSparseMatrixCSC{T,M}, B::CuSparseMatrixCSC{T,M}) where {T,M}
+    A2 = CuSparseMatrixCSR(A)
+    B2 = CuSparseMatrixCSR(B)
+    CuSparseMatrixCSC(gemm('N', 'N', one(T), A2, B2, 'O'))
+end
+function LinearAlgebra.:(*)(A::CuSparseMatrixCOO{T,M}, B::CuSparseMatrixCOO{T,M}) where {T,M}
+    A2 = CuSparseMatrixCSR(A)
+    B2 = CuSparseMatrixCSR(B)
+    CuSparseMatrixCOO(gemm('N', 'N', one(T), A2, B2, 'O'))
 end
 
-function Base.:(-)(A::CuSparseMatrixCSR, B::CuSparseMatrix)
-    csrB = CuSparseMatrixCSR(B)
-    return geam(one(eltype(A)), A, -one(eltype(A)), csrB, 'O')
+function SparseArrays.droptol!(A::CuSparseMatrixCOO{T,M}, tol::Real) where {T,M}
+    mask = abs.(A.nzVal) .> tol
+    rows = A.rowInd[mask]
+    cols = A.colInd[mask]
+    datas = A.nzVal[mask]
+    B = sparse(rows, cols, datas, size(A)..., fmt = :coo)
+    copyto!(A, B)
 end
 
-function Base.:(+)(A::CuSparseMatrix, B::CuSparseMatrixCSR)
-    csrA = CuSparseMatrixCSR(A)
-    return geam(one(eltype(A)), csrA, one(eltype(A)), B, 'O')
+for SparseMatrixType in [:CuSparseMatrixCSC, :CuSparseMatrixCSR, :CuSparseMatrixCOO]
+    @eval begin
+        if $SparseMatrixType in [CuSparseMatrixCSC, CuSparseMatrixCSR]
+
+            Base.reshape(A::$SparseMatrixType{T,M}, dims::NTuple{N,Int}) where {T,N,M} = 
+            $SparseMatrixType( reshape(CuSparseMatrixCOO(A), dims) )
+
+            function SparseArrays.droptol!(A::$SparseMatrixType{T,M}, tol::Real) where {T,M}
+                B = copy(CuSparseMatrixCOO(A))
+                droptol!(B, tol)
+                copyto!(A, $SparseMatrixType(B))
+            end
+            
+        end
+
+        LinearAlgebra.:(*)(A::Transpose{T,<:$SparseMatrixType}, B::$SparseMatrixType{T,M}) where {T,M} = _sptranspose(parent(A)) * B
+        LinearAlgebra.:(*)(A::Transpose{T,<:$SparseMatrixType}, B::Transpose{T,<:$SparseMatrixType}) where {T} = _sptranspose(parent(A)) * _sptranspose(parent(B))
+        LinearAlgebra.:(*)(A::$SparseMatrixType{T,M}, B::Transpose{T,<:$SparseMatrixType}) where {T,M} = A * _sptranspose(parent(B))
+        LinearAlgebra.:(*)(A::Adjoint{T,<:$SparseMatrixType}, B::$SparseMatrixType{T,M}) where {T,M} = _spadjoint(parent(A)) * B
+        LinearAlgebra.:(*)(A::Adjoint{T,<:$SparseMatrixType}, B::Adjoint{T,<:$SparseMatrixType}) where {T} = _spadjoint(parent(A)) * _spadjoint(parent(B))
+        LinearAlgebra.:(*)(A::$SparseMatrixType{T,M}, B::Adjoint{T,<:$SparseMatrixType}) where {T,M} = A * _spadjoint(parent(B))
+    end
 end
 
-function Base.:(-)(A::CuSparseMatrix, B::CuSparseMatrixCSR)
-    csrA = CuSparseMatrixCSR(A)
-    return geam(one(eltype(A)), csrA, -one(eltype(A)), B, 'O')
+function _spadjoint(A::CuSparseMatrixCSR{T,M}) where {T,M}
+    cscA = CuSparseMatrixCSC(conj(A))
```

Comment:
```suggestion
    cscA = CuSparseMatrixCSC(A.rowPtr, A.colVal, conj(A.nzVal), reverse(size(A))
```

Reply:
Why should this be better than the previous implementation?

Reply:
it will avoid to copy `A.rowPtr` and `A.colVal`.